lightning/src/util/ser_macros.rs

```diff
@@ -173,13 +174,28 @@ macro_rules! impl_writeable {
 					if $len != 0 {
 						use util::ser::LengthCalculatingWriter;
 						let mut len_calc = LengthCalculatingWriter(0);
-						$( self.$field.write(&mut len_calc)?; )*
+						$( self.$field.write(&mut len_calc).expect("No in-memory data may fail to serialize"); )*
 						assert_eq!(len_calc.0, $len);
 					}
 				}
 				$( self.$field.write(w)?; )*
 				Ok(())
 			}
+
+			#[inline]
+			fn serialized_length(&self) -> usize {
+				if $len == 0 || cfg!(any(test, feature = "fuzztarget")) {
+					let mut len_calc = 0;
+					$( len_calc += self.$field.serialized_length(); )*
+					if $len != 0 {
+						// In tests, assert that the hard-coded length matches the actual one
+						assert_eq!(len_calc, $len);
```

Comment:
Would this be covered during fuzzing? Seems like a useful check but codecov thinks it isn't hit

Reply:
Good catch!